bootstrap/src/sun/nio/ch/lincheck/EventTracker.java

```diff
@@ -81,6 +81,11 @@ void cacheInvokeDynamicCallSite(
 
     BootstrapResult<?> invokeDeterministicallyOrNull(long descriptorId, Object descriptor, Object receiver, Object[] params);
 
+    void beforeLoopEnter(int codeLocation, int loopId);
+    void onLoopIteration(int codeLocation, int loopId);
+    void afterLoopExit(int codeLocation, int loopId);
+    void afterLoopExceptionExit(int codeLocation, int loopId, Throwable exception, boolean canEnterFromOutsideLoop);
+
```

Comment:
Should be part of the separate PR, not here. See Nikita's PR https://github.com/JetBrains/lincheck/pull/800